### ISSUE TITLE
SA1119: False positive with switch expression

### ISSUE THREAD

**Issue by neonb** [DotNetAnalyzers/StyleCopAnalyzers#3520]
Directly using the value that a _switch expression_ evaluates to requires the use of parentheses:
```
var input = 5;
var output = (input switch
    {
        1 => "one",
        _ => "some other number",
    })
    .ToUpper();
```
StyleCop reports the parentheses as unnecessary, even though removing them results in a compiler error.

(Tested in JetBrains Rider by adding `StyleCop.Analyzers` from NuGet to an empty project – I hope Rider's own features aren't conflicting here.)

**Comment by ZacharyPatten** [comment]
What version of the package were you using? If you are using `1.1.118` then you will see a warning, but if you use the latest prerelease `1.2.0-beta.435` it is no longer warning for me.
